src/Pim/Component/Catalog/Builder/ProductTemplateBuilder.php

```diff
@@ -64,55 +42,42 @@ public function createProductTemplate()
     /**
      * {@inheritdoc}
      */
-    public function addAttributes(ProductTemplateInterface $template, array $attributes, $locale)
+    public function addAttributes(ProductTemplateInterface $template, array $attributes)
     {
-        $options = [
-            'entity'                     => 'product',
-            'locale'                     => $locale,
-            'disable_grouping_separator' => true
-        ];
-
-        $values = $this->buildProductValuesFromTemplateValuesData($template, $attributes, $locale);
-        $valuesData = $this->normalizer->normalize($values, 'standard', $options);
-        $template->setValuesData($valuesData);
+        $values = $this->buildProductValuesFromTemplateValuesData($template, $attributes);
+        $template->setValues($values);
     }
 
     /**
      * {@inheritdoc}
      */
     public function removeAttribute(ProductTemplateInterface $template, AttributeInterface $attribute)
     {
-        $valuesData = $template->getValuesData();
+        $values = $template->getValues();
 
-        unset($valuesData[$attribute->getCode()]);
+        foreach ($values as $value) {
```

Comment:
shouldn't we add a method in the collection for that?
it seems to be a regular case :)